src/components/hero/AcceptDemoInvitation.tsx

```diff
@@ -0,0 +1,106 @@
+import { Box } from '@mui/material';
+import { PostgrestError } from '@supabase/postgrest-js';
+import { submitDirective } from 'api/directives';
+import useDirectiveGuard from 'app/guards/hooks';
+import MessageWithButton from 'components/content/MessageWithButton';
+import Error from 'components/shared/Error';
+import { jobStatusQuery, trackEvent } from 'directives/shared';
+import { Dispatch, SetStateAction, useState } from 'react';
+import { jobStatusPoller } from 'services/supabase';
+import { useEntitiesStore_mutate } from 'stores/Entities/hooks';
+
+interface Props {
+    tenant: string;
+    loading: boolean;
+    setLoading: Dispatch<SetStateAction<boolean>>;
+    setOpen: Dispatch<SetStateAction<boolean>>;
+    goToFilteredTable: () => void;
+}
+
+const directiveName = 'acceptDemoTenant';
+
+function AcceptDemoInvitation({
+    tenant,
+    loading,
+    setLoading,
+    setOpen,
+    goToFilteredTable,
+}: Props) {
+    const { directive, mutate } = useDirectiveGuard(directiveName, {
+        hideAlert: true,
+    });
+
+    const mutateAuthRoles = useEntitiesStore_mutate();
+
+    const [serverError, setServerError] = useState<PostgrestError | null>(null);
+
+    const applyDirective = async (): Promise<void> => {
+        if (directive) {
+            setLoading(true);
+
+            const response = await submitDirective(
+                directiveName,
+                directive,
+                tenant
+            );
+
+            if (response.error) {
+                setLoading(false);
+                return setServerError(response.error as PostgrestError);
+            }
+
+            const data = response.data[0];
+
+            jobStatusPoller(
```

Comment:
Not for this pr... we need to take another look and see if we can get more directive stuff shared. However, I do think it might be the case where they're all _similar_ but not _identical_ and would be a bigger pain than it is worth.